test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/SkipCriteria.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.rest.yaml.section;
+
+public interface SkipCriteria {
```

Comment:
I think the skip decision is a good idea, but I kept `testFeatures` separate for another reason; test features are "static" meta features supported by the test runner, not the cluster. It's a semi-fixed (mostly at compile time) set of "things" that the test runner (not the cluster under examination) must support, or the test is not run. That's why I wanted to keep this separate.
(Also, I suspect this was more used in the past, and now it's mainly used to test xpack/no_xpack).

Maybe I should rename this as to include `Cluster` somewhere? `ClusterSkipCriteria`?

Reply:
Re: keeping it as is, the reason I did this PR is that I was introducing a skip criteria based on features, and cramming it into the current framework was definitely not "clean"; you can already see the thing was designed with just version of test features in mind, and OS was retrofitted later.
I wanted to avoid adding another separate check (with unclear relationships with the existing ones - does it run before or after?) and make things more explicit, and make this change so now there is one `skip` check which does all the checks, built incrementally with composable criteria from parsing.

Reply:
Oh, I get it... this part isn't related to the feature api at all. Sorry, that was confusing but makes sense now. How about evaluating the testFeature check in the builder to make that clear? Your builder could return an overridden `SkipAlwaysSection` in that case while `SkipSection` is only based on skip criteria?

Reply:
Unfortunately SkipSection already exposes to some extent test features (for cross checking/validation). We can work to make that better, but I think it's for another PR. Meanwhile I reduced it to just a test method (yamlRunnerHasFeature), instead of esposing the whole list, and still used your idea to inject a single always-true skip criteria.